gencode.c

```diff
@@ -1511,7 +1544,7 @@ gen_load_llrel(offset, size)
 		 * do an indirect load, to use the X register as an offset.
 		 */
 		s2 = new_stmt(BPF_LD|BPF_IND|size);
-		s2->s.k = offset;
+		s2->s.k = offset + off_ll;
```

Comment:
"off_ll", in this case, is presumably not the offset of the beginning of the link-layer header from the beginning of the raw packet data, as said offset is in the X register.  A different variable should be used there, even if that means that a new variable is introduced.